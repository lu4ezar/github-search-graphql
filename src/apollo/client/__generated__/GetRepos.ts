/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: GetRepos
// ====================================================

export interface GetRepos_search_edges_node_App {
  __typename: "App" | "Issue" | "MarketplaceListing" | "Organization" | "PullRequest" | "User";
}

export interface GetRepos_search_edges_node_Repository_stargazers {
  __typename: "StargazerConnection";
  /**
   * Identifies the total count of items in the connection.
   */
  totalCount: number;
}

export interface GetRepos_search_edges_node_Repository_watchers {
  __typename: "UserConnection";
  /**
   * Identifies the total count of items in the connection.
   */
  totalCount: number;
}

export interface GetRepos_search_edges_node_Repository {
  __typename: "Repository";
  /**
   * The repository's URL.
   */
  homepageUrl: any | null;
  /**
   * The name of the repository.
   */
  name: string;
  /**
   * The description of the repository.
   */
  description: string | null;
  /**
   * The description of the repository rendered to HTML.
   */
  descriptionHTML: any;
  /**
   * The HTTP URL for this repository
   */
  url: any;
  /**
   * A list of users who have starred this starrable.
   */
  stargazers: GetRepos_search_edges_node_Repository_stargazers;
  /**
   * A list of users watching the repository.
   */
  watchers: GetRepos_search_edges_node_Repository_watchers;
}

export type GetRepos_search_edges_node = GetRepos_search_edges_node_App | GetRepos_search_edges_node_Repository;

export interface GetRepos_search_edges {
  __typename: "SearchResultItemEdge";
  /**
   * The item at the end of the edge.
   */
  node: GetRepos_search_edges_node | null;
  /**
   * A cursor for use in pagination.
   */
  cursor: string;
}

export interface GetRepos_search {
  __typename: "SearchResultItemConnection";
  /**
   * A list of edges.
   */
  edges: (GetRepos_search_edges | null)[] | null;
}

export interface GetRepos {
  /**
   * Perform a search across resources.
   */
  search: GetRepos_search;
}

export interface GetReposVariables {
  searchString: string;
  after?: string | null;
}
