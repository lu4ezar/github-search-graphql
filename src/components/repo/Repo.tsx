import React from "react";
import Octicon, {
  Globe,
  Eye,
  MarkGithub,
  Star,
  CircleSlash
} from "@primer/octicons-react";
import { StyledRepo, Text, Link, IconContainer, Icon } from "./styled";
// eslint-disable-next-line camelcase
import { GetRepos_search_edges_node_Repository } from "../../apollo/client/__generated__/GetRepos";

const Repo = ({
  homepageUrl,
  name,
  description,
  url,
  stargazers,
  watchers
}: // eslint-disable-next-line camelcase
GetRepos_search_edges_node_Repository) => (
  <StyledRepo>
    <Text>
      <h3>
        <Link
          href={homepageUrl || url}
          title="open homepage in new tab"
          target="_blank"
          rel="noopener noreferrer"
        >
          {name}
        </Link>
      </h3>
      <p>{description || ""}</p>
    </Text>
    <IconContainer>
      <Icon>
        {homepageUrl ? (
          <>
            <Link
              title="homepage"
              href={homepageUrl}
              target="_blank"
              rel="noopener noreferrer"
            >
              <Octicon icon={Globe} size="medium" verticalAlign="middle" />
            </Link>
            <div>Homepage</div>
          </>
        ) : (
          <div className="not-allowed" title="no homepage">
            <Octicon icon={CircleSlash} size="medium" verticalAlign="middle" />
            <div>No Homepage</div>
          </div>
        )}
      </Icon>
      <Icon>
        <Link
          title="github page"
          href={url}
          target="_blank"
          rel="noopener noreferrer"
        >
          <Octicon icon={MarkGithub} size="medium" verticalAlign="middle" />
        </Link>
        <div>Github</div>
      </Icon>
      <Icon title="stargazers">
        <Octicon icon={Star} size="medium" verticalAlign="middle" />
        <div>{stargazers.totalCount}</div>
      </Icon>
      <Icon title="watchers">
        <Octicon icon={Eye} size="medium" verticalAlign="middle" />
        <div>{watchers.totalCount}</div>
      </Icon>
    </IconContainer>
  </StyledRepo>
);

export default Repo;
