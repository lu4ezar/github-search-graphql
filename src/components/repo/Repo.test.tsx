/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { render } from "@testing-library/react";
import Repo from "./Repo";
import { GetRepos_search_edges_node_Repository as RepoProps } from "../../apollo/client/__generated__/GetRepos";

const repoProps: RepoProps = {
  __typename: "Repository",
  homepageUrl: null,
  name: "github-search-graphql",
  description: "Now with Github API v4! (graphql)",
  url: "https://github.com/lu4ezar/github-search-graphql",
  stargazers: {
    __typename: "StargazerConnection",
    totalCount: 0
  },
  watchers: {
    __typename: "UserConnection",
    totalCount: 1
  }
};

const renderComponent = ({ ...additionalProps } = {}) =>
  render(<Repo {...repoProps} {...additionalProps} />);

it("matches the snapshot", () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();
});

it("can show 'No description was provided' message", () => {
  const description = null;
  const { getByText } = renderComponent({ description });
  const message = getByText("No description was provided");
  expect(message).toBeInTheDocument();
});

it("shows Homepage link", () => {
  const homepageUrl = "https://www.google.com";
  const { getByTitle } = renderComponent({ homepageUrl });
  const link = getByTitle("homepage");
  expect(link).toHaveAttribute("href", homepageUrl);
});
