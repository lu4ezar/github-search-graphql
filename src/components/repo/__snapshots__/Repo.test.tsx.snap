// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm ccikoI"
    role="listitem"
  >
    <div
      class="sc-AxirZ eiccaF"
    >
      <h3>
        <a
          class="sc-AxhUy ffpEsK"
          href="https://github.com/lu4ezar/github-search-graphql"
          rel="noopener noreferrer"
          target="_blank"
          title="open homepage in new tab"
        >
          github-search-graphql
        </a>
      </h3>
      <p
        class="description"
      >
        Now with Github API v4! (graphql)
      </p>
    </div>
    <div
      class="sc-AxiKw hTACOn"
    >
      <div
        class="sc-AxhCb gwkJWS"
      >
        <div
          class="not-allowed"
          title="no homepage"
        >
          <svg
            aria-hidden="true"
            class="octicon"
            height="32"
            role="img"
            style="display: inline-block; fill: currentColor; user-select: none; vertical-align: middle;"
            viewBox="0 0 14 16"
            width="28"
          >
            <path
              d="M7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm0 1.3c1.3 0 2.5.44 3.47 1.17l-8 8A5.755 5.755 0 011.3 8c0-3.14 2.56-5.7 5.7-5.7zm0 11.41c-1.3 0-2.5-.44-3.47-1.17l8-8c.73.97 1.17 2.17 1.17 3.47 0 3.14-2.56 5.7-5.7 5.7z"
              fill-rule="evenodd"
            />
          </svg>
          <div>
            No Homepage
          </div>
        </div>
      </div>
      <div
        class="sc-AxhCb gwkJWS"
      >
        <a
          class="sc-AxhUy ffpEsK"
          href="https://github.com/lu4ezar/github-search-graphql"
          rel="noopener noreferrer"
          target="_blank"
          title="github page"
        >
          <svg
            aria-hidden="true"
            class="octicon"
            height="32"
            role="img"
            style="display: inline-block; fill: currentColor; user-select: none; vertical-align: middle;"
            viewBox="0 0 16 16"
            width="32"
          >
            <path
              d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
        <div>
          Github
        </div>
      </div>
      <div
        class="sc-AxhCb gwkJWS"
        title="stargazers"
      >
        <svg
          aria-hidden="true"
          class="octicon"
          height="32"
          role="img"
          style="display: inline-block; fill: currentColor; user-select: none; vertical-align: middle;"
          viewBox="0 0 14 16"
          width="28"
        >
          <path
            d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74L14 6z"
            fill-rule="evenodd"
          />
        </svg>
        <div>
          0
        </div>
      </div>
      <div
        class="sc-AxhCb gwkJWS"
        title="watchers"
      >
        <svg
          aria-hidden="true"
          class="octicon"
          height="32"
          role="img"
          style="display: inline-block; fill: currentColor; user-select: none; vertical-align: middle;"
          viewBox="0 0 16 16"
          width="32"
        >
          <path
            d="M8.06 2C3 2 0 8 0 8s3 6 8.06 6C13 14 16 8 16 8s-3-6-7.94-6zM8 12c-2.2 0-4-1.78-4-4 0-2.2 1.8-4 4-4 2.22 0 4 1.8 4 4 0 2.22-1.78 4-4 4zm2-4c0 1.11-.89 2-2 2-1.11 0-2-.89-2-2 0-1.11.89-2 2-2 1.11 0 2 .89 2 2z"
            fill-rule="evenodd"
          />
        </svg>
        <div>
          1
        </div>
      </div>
    </div>
  </div>
</div>
`;
